packages/types/src/Metadata/v6/Constants.ts

```diff
@@ -0,0 +1,48 @@
+// Copyright 2017-2019 @polkadot/types authors & contributors
+// This software may be modified and distributed under the terms
+// of the Apache-2.0 license. See the LICENSE file for details.
+
+import Bytes from '../../primitive/Bytes';
+import Text from '../../primitive/Text';
+import Type from '../../primitive/Type';
+import Struct from '../../codec/Struct';
+import Vector from '../../codec/Vector';
+
+export class ModuleConstantMetadata extends Struct {
+  constructor (value?: any) {
+    super({
+      name: Text,
+      ty: Text,
```

Comment:
Here, it can be `ty: Type`, which is a subclass of Text which deals with rust types.

Reply:
And actually we use `type: Type` in JS, it's not the same as Rust, but ty is just weird

Reply:
true, i was matching directly with whatever is in Rust, plus type is a reserved keyword so to me that was more confusing 😄